File: src/main/webapp/storage/src/app/dto/MovingRecord.ts
import {Item} from "./Item";

export class MovingRecord {
  private readonly _id: number;
  private readonly _count: number;
  private readonly _date: string;
  private readonly _type: string;
  private readonly _item: Item;


  constructor(id: number, count: number, date: string, type: string, item: Item) {
    this._id = id;
    this._count = count;
    this._date = date;
    this._type = type;
    this._item = item;
  }


  get id(): number {
    return this._id;
  }

  get count(): number {
    return this._count;
  }

  get date(): string {
    return this._date;
  }

  get type(): string {
    return this._type;
  }

  get item(): Item {
    return this._item;
  }
}

File: src/main/java/com/sonet/storage/model/moving/MovingRecord.java
package com.sonet.storage.model.moving;

import com.sonet.storage.model.item.Item;
import com.sonet.storage.model.user.User;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;

@Entity
@Table(name = "moving_record")
@NoArgsConstructor
@ToString
@Getter
@Setter
public class MovingRecord {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long count;

    private String date;

    @ManyToOne
    private User user;

    @OneToOne
    @JoinTable(  name = "moving_record_type",
            joinColumns = @JoinColumn(name = "record_id"),
            inverseJoinColumns = @JoinColumn(name = "type_id"))
    private MovingType type;

    @OneToOne
    private Item item;
}
